src/entities/marking/api/getTemporaryMarkingList.ts

```diff
@@ -0,0 +1,103 @@
+import { skipToken, useInfiniteQuery } from "@tanstack/react-query";
+import { Nickname, UserId } from "@/entities/profile/api";
+import { apiClient } from "@/shared/lib";
+import { useAuthStore } from "@/shared/store";
+import { MY_MARKING_ENDPOINT } from "../constants";
+
+export interface TempMarkingFileInfo {
+  id: number;
+  imageUrl: string;
+  lank: number;
+  regDt: string;
+}
+// TODO 타입 스크립트 리팩토링 시 변경 하기
+export interface TempMarkingInfo {
+  markingId: number;
+  region: string;
+  content: string | null;
+  isVisible: "PUBLIC" | "FOLLOWERS_ONLY" | "PRIVATE";
+  regDt: string;
+  previewImage: string | null;
+  userId: UserId;
+  nickname: Nickname;
+  isOwner: true;
+  isTempSaved: true;
+  lat: number;
+  lng: number;
+  address: {
+    id: number;
+    province: string;
+    cityCounty: string;
+    district: string | null;
+    subDistrict: string;
+  };
+  countData: {
+    likedCount: 0;
+    savedCount: 0;
+  };
+  images: TempMarkingFileInfo[];
+}
+
+export interface GetTemporaryMarkingListResponse {
+  markings: TempMarkingInfo[];
+  totalElements: number;
+  totalPages: number;
+  pageAble: {
+    pageNumber: number;
+    pageSize: number;
+    sort: {
+      sorted: boolean;
+      unsorted: boolean;
+      empty: boolean;
+    };
+  };
+  offset: number;
+  paged: boolean;
+  unpaged: boolean;
+}
+
+export const useGetTemporaryMarkingList = () => {
+  const token = useAuthStore((state) => state.token);
+  return useInfiniteQuery({
+    queryKey: ["temporaryMarkingList"],
+    queryFn: token
+      ? ({ pageParam }: { pageParam: number }) =>
+          apiClient.get<GetTemporaryMarkingListResponse>(
+            MY_MARKING_ENDPOINT.TEMPORARY(pageParam),
+            {
+              withToken: true,
+            },
+          )
+      : skipToken,
+    getNextPageParam: ({ totalPages, pageAble }) => {
+      return pageAble.pageNumber < totalPages - 1
+        ? pageAble.pageNumber + 1
+        : null;
+    },
+    initialPageParam: 0,
+    /**
+     * 페이징 된 데이터를 하나의 객체로 만든 후 엔트리 형태로 변환하여 반환 합니다.
+     * [날짜 , TemporaryMarkingInfo[]] 형태로 반환 합니다.
+     */
+    select: ({ pages }) => {
+      const markings = pages.flatMap((page) => page.markings);
+      const temporaryMarkingMap = markings.reduce<
+        Record<string, TempMarkingInfo[]>
+      >((map, { regDt, ...rest }) => {
+        const year = new Date(regDt).getFullYear();
+        const month = `${new Date(regDt).getMonth() + 1}`.padStart(2, "0");
+        const day = `${new Date(regDt).getDate()}`.padStart(2, "0");
+        const key = `${year}.${month}.${day}`;
```

Comment:
이미 yy.mm.dd 형태로 바꿔주는 util 함수가 있습니다.
shared/lib에 있는 `formatDateToYearMonthDay` 이용해주시면 됩니다.

```suggestion
const key = formatDateToYearMonthDay(regDt)
```